gluten-core/src/main/scala/io/glutenproject/expression/ArrayExpressionTransformer.scala

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.glutenproject.expression
+
+import io.glutenproject.expression.ConverterUtils.FunctionConfig
+import io.glutenproject.substrait.`type`.TypeBuilder
+import io.glutenproject.substrait.expression.{ExpressionBuilder, ExpressionNode}
+
+import org.apache.spark.internal.Logging
+import org.apache.spark.sql.catalyst.expressions._
+import org.apache.spark.sql.types.DataType
+
+import scala.collection.mutable.ArrayBuffer
+
+class CreateArrayTransformer(children: Seq[ExpressionTransformer], useStringTypeWhenEmpty: Boolean,
+  original: CreateArray)
+    extends ExpressionTransformer
+    with Logging {
+
+  override def doTransform(args: java.lang.Object): ExpressionNode = {
+    val childNodes = new java.util.ArrayList[ExpressionNode]()
+    children.foreach(child => {
+      val childNode = child.doTransform(args)
+      childNodes.add(childNode)
+    })
+
+    // Ignore useStringTypeWhenEmpty for clickhouse backend
```

Comment:
does Velox backend need? @rui-mo

Reply:
I find in Spark this config determines the element type to be NullType or StringType when args are empty. In Velox, the NullType way is used by default, but I think the backend needs to follow what Spark plans. Maybe we can send this config to native and mark the function option as `req`.

Reply:
ClickHouse has the same behavior with Velox.  
```
:) select array() as x,  toTypeName(x);

SELECT
    [] AS x,
    toTypeName(x)

Query id: 73ec966f-00a6-49b7-9936-77e44ce17f14

┌─x──┬─toTypeName(array())─┐
│ [] │ Array(Nothing)      │
└────┴─────────────────────┘

1 rows in set. Elapsed: 0.002 sec. 
```

Reply:
@rui-mo I think currently we can partially support transforming to substrait plan only when `useStringTypeWhenEmpty` is false, before Velox and ClickHouse supports `useStringTypeWhenEmpty`,

Reply:
Sounds good to me.